source/computer/commands.rst

```diff
@@ -166,44 +166,43 @@ find
     # 查找 ~/src 目录及其子目录下以 .c 结尾的文件
     $ find ~/src -type f -name "*.c"
 
+使用 ``-delete`` 选项可以直接删除查找的文件或目录::
+
     # 查找 ~/src 目录及其子目录下以 .pyc 结尾的文件，并删除
     $ find ~/src -type f -name "*.pyc" -delete
 
-    # 查找 ~/src 目录及其子目录下以 .c 结尾的文件，并使用 grep 命令找出这些文件中含 seismo-learn 的行
-    $ find ~/src -type f -name "*.c" -exec grep seismo-learn {} +
+``find`` 的 ``-exec`` 选项可以调用其他系统命令直接对查找的结果进行处理::
 
-最后一个命令借助 ``-exec`` 选项，与 `grep`_ 命令结合使用。其中 ``{}`` 用来与 ``-exec`` 选项结合
-使用，表示查找到的文件。借助该选项，``find`` 命令可以跟许多其他 Linux 的命令结合使用。
+    # 查找 ~/src 目录及其子目录下以 .c 结尾的文件，并执行 grep 命令找出这些文件中含 seismo-learn 的行
+    $ find ~/src -type f -name "*.c" -exec grep seismo-learn {} +
 
+这个例子中，``{}`` 与 ``-exec`` 选项结合，表示查找到的文件。
 我们还可以用 ``-ok`` 来代替 ``-exec`` 选项，二者的区别是 ``-ok`` 选项在执行后面的命令前会给出提示，
 输入 :kbd:`y` 才会执行，输入 :kbd:`n` 则不执行。
 
 gawk
 ----
 
-``awk`` 命令的命名起源于其三位作者的姓氏首字母。该命令可以选择标准输入、其它命令的输出或文件中
-的特定字段并进行操作。它依次扫描每一行，并读取里面的每一个字段。可以参考 
-`awk 入门教程 <https://www.ruanyifeng.com/blog/2018/11/awk.html>`__ 学习其用法。
-
-``gawk`` 是 GNU 版本的 ``awk`` 命令。Linux 系统下的 ``awk`` 命令一般是指向
-``gawk`` 命令的软链接，可以使用以下命令查看::
+``awk`` 命令的命名起源于其三位作者的姓氏首字母。该命令可以选择标准输入、其他命令的输出或文件中
+的特定字段并进行操作。它依次扫描每一行，并读取里面的每一个字段。可以参考
+`awk 入门教程 <https://www.ruanyifeng.com/blog/2018/11/awk.html>`__ 学习其更多用法。
 
-    $ ls -l $(which awk)
-    lrwxrwxrwx. 1 root root 4 Aug  4  2018 /usr/bin/awk -> gawk
+``gawk`` 是 GNU 版本的 ``awk`` 命令。通常建议直接使用 ``gawk`` 而非 ``awk``，
+尽管在 Linux 系统下，``awk`` 命令一般是指向 ``gawk`` 命令的软链接。
 
 ::
 
-    # 将一个文件的每一行输出到标准输出
+    # 将一个文件的每一行输出到标准输出。 $0 表示当前行
     $ gawk '{print $0}' file
 
     # 将一个文件的每一行的第 1 个和第 3 个字段输出到标准输出（字段的默认分隔符是空格和制表符）
     $ gawk '{print $1,$3}' file
 
-    # 同上，但忽略第 1 行和第 2 行，从第 3 行开始
+    # 同上，但跳过第 1-2 行，从第 3 行开始。FNR 为当前行的行数
```

Comment:
I am not sure the difference between NR and FNR

Reply:
It's different if you're working on multiple files.

Reply:
Soga.

Reply:
So NR will continously count the number of row, right?

Reply:
I think so.